index.bs

```diff
@@ -1294,44 +1296,114 @@ A {{ReceiveStream}} has the following internal slots.
-      1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-         from.
-      1. Let |reason| be the value from the message from the remote side.
-      1. [=ReadableStream/Error=] |stream| with |reason|.
-
-The {{ReceiveStream}}'s <dfn>cancelAlgorithm</dfn> is:
-     1. Let |stream| be the {{ReceiveStream}} object.
-     1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-        from.
-     1. Let |reason| be the first argument, if it is numeric, or 0.
-     1. If |stream| has received a RESET_STREAM frame, abort these steps.
-     1. [=In parallel=], [=send STOP_SENDING=] over |stream|'s
-        [=WebTransport stream=] with |reason| as the Application Protocol Error
-         Code.
+<div algorithm>
+
+To <dfn for="ReceiveStream">pull bytes</dfn> from a {{ReceiveStream}} |stream|, run these steps.
+
+1. Let |transport| be |stream|'s [=ReceiveStream/[[Transport]]=].
+1. Let |internalStream| be |stream|'s [=ReceiveStream/[[InternalStream]]=].
+1. Let |promise| be a new promise.
+1. Let |buffer|, |offset|, and |maxBytes| be null.
+1. If |stream|'s [=ReadableStream/current BYOB request view=] for |stream| is not null:
+  1. Set |offset| to |stream|'s [=ReadableStream/current BYOB request view=].\[[ByteOffset]].
+  1. Set |maxBytes| to |stream|'s [=ReadableStream/current BYOB request view=]'s
+     [=BufferSource/byte length=].
+  1. Set |buffer| to |stream|'s [=ReadableStream/current BYOB request view=]'s
+     [=BufferSource/underlying buffer=].
+1. Otherwise:
+  1. Set |offset| to 0.
+  1. Set |maxBytes| to an [=implementation-defined=] size.
+  1. Set |buffer| be a [=new=] {{ArrayBuffer}} with |maxBytes| size. If allocating the
+     {{Uint8Array}} fails, return a promise [=rejected with=] a {{RangeError}}.
+1. Return |promise| and run the remaining steps [=in parallel=].
+1. [=ArrayBuffer/Write=] the bytes that area [=stream/receive|read=] from |internalStream| into
+   |buffer| with offset |offset|, up to |maxBytes| bytes. Wait until either at least one byte is
+   read or FIN is received. Let |read| be the number of read bytes, and let |hasReceivedFIN| be
+   whether FIN was accompanied.
+   Note: The user-agent MAY have a buffer to improve the transfer performance. Such a buffer
+   SHOULD have a fixed size, to carry the backpressure information to the server.
+
+   Note: This operation may return before filling up all of |bytes|.
+
+1. If the previous step failed, abort the remaining steps.
+
+  Note: We don't reject |promise| here because we handle network errors elsewhere, and those logic
+  error |stream| and reject the result of this read operation.
+
+1. [=Queue a network task=] with |transport| to run these steps:
+
+   Note: If the buffer described above is available in the [=event loop=] where this procedure is
+   running, the following steps may run immediately.
+
+  1. If |read| > 0:
+    1. Set |view| to a [=new=] {{Uint8Array}} with |buffer|, |offset| and |read|.
```

Comment:
Nit: WebIDL's `new` doesn't call the constructor. I think you need to do something like
```
1. Set |view| to ! Construct(constructor, « |buffer|, |offset|, |read| »).
```
as in https://heycam.github.io/webidl/#arraybufferview-create

Reply:
I would suggest instead just removing the link so it's `new {{Uint8Array}}`. This is IMO a missing primitive that Web IDL should supply, similar to how it supplies https://heycam.github.io/webidl/#arraybufferview-create .

It's better to avoid reaching into ES spec internals when we can.

Reply:
Thank you, done.